jms/src/main/scala/akka/stream/alpakka/jms/JmsMessages.scala

```diff
@@ -7,10 +7,34 @@ package akka.stream.alpakka.jms
 import scala.collection.JavaConverters._
 import java.util
 
-sealed trait JmsHeader
-final case class JmsCorrelationId(jmsCorrelationId: String) extends JmsHeader
-final case class JmsReplyTo(jmsDestination: Destination) extends JmsHeader
-final case class JmsType(jmsType: String) extends JmsHeader
+sealed trait JmsHeader {
+
+  /**
+   * Indicates if this header must be set during the send() operation according to the JMS specification or as attribute of the jms message before.
+   */
+  def usedDuringSend(): Boolean
+}
+
+final case class JmsCorrelationId(jmsCorrelationId: String) extends JmsHeader {
+  override def usedDuringSend(): Boolean = false
```

Comment:
when declared without the () in the trait, you may override it with val

Reply:
Fixed